### ISSUE TITLE
Match container name by regex

### ISSUE THREAD

**Issue by danehans** [discordianfish/docker-spotter#8]
I am trying to match on container names that include a certain pattern. For example, execute foo.sh whenever you see a container with a name that includes _bar_. Is that possible? If not, is it possible to match on the container's hostname?

**Comment by discordianfish** [comment]
@danehans: No, sorry - that's not possible right now. Although if someone would implement this, I'm happy to merge it. The naming stuff could be easily added by using a regexp to match the container name instead of a string.

**Comment by dreamcat4** [comment]
Hello. Need this to invoke a command on any container start event… Perhaps implementing this is possible without too much fuss? So far have not seen any other such tools who can do this.

**Comment by danehans** [comment]
@dreamcat4 I achieved the functionality I was looking for but using ENV var's instead of a container name. For example, I have docker-spotter kick off a pipework command whenever it sees the ENV CONFIG_NETWORK=TRUE I found this implementation to be more flexible:

https://github.com/danehans/docker-spotter/commit/9167d228baa213146ca0b0829cbba96957077452

I believe I needed to clean up my commits to get these changes up stream. If you find that my implementation meets your needs, do you mind pushing the code upstream?

**Comment by dreamcat4** [comment]
@danehans Sorry I can't understand Go very well enough to peer-review your changes or suggest improvements. But I was under the impression that it is *only* at this 1 specific line that you need to swap for regexp matching function:

https://github.com/discordianfish/docker-spotter/blob/master/spotter.go#L174

This code is a wee bit too obscure for me, not knowing any Go whatsoever:

	events = hm[strings.TrimLeft(container.Name, "/")]
	if events == nil {
		continue
	}

Anyways. In the meantime, (to solve my own immediate needs) I've come up with this temporary solution in shell script:

    docker events --filter='event=start' |  \
    while read container_id; do
        container_id=$(echo $continer_id | cut -d ' ' -f2)
        container_json_metadata="$(docker inspect ${container_id%:})"
        # do work here
    done

The key difference being that it is triggered on *any* container start events, rather than only those start events with a specific container name.

It seems to work well enough, although somewhat slower than compiled go program.

**Comment by discordianfish** [comment]
@danehans I live the env var based search, happy to merge that if you send a PR!

@dreamcat4 Someone already implemented wildcard support. I've just merged that PR, does that solve your problem?

**Comment by dreamcat4** [comment]
@discordianfish Yes. Simple wildcard `*` works now to pick up all container names. Which is enough for me. Although of course a regexp would be more powerful, I don't happen to need regexp myself.

The cmd i ran was:

    $ ./docker-spotter -since=0 -e '*:start:echo:{{.Name}}'
    2015/02/04 12:26:43 = *:start:*:start:echo:{{.Name}}
    2015/02/04 12:27:20 > /bin/echo [ [/tvheadend] ]
    2015/02/04 12:27:20 - /tvheadend

BTW: A small problem with my shell script's `while read ...` loop is that it is invoked as a sub shell / sub-process. Which means that any variables= being set inside the `while read …` loop are not remembered and visible further down in the shell script after the while loop has completed.

**Comment by discordianfish** [comment]
@dreamcat4 Right, matching via regexp seems like a useful feature, so if someone wants to implement that, I'm happy to merge it. I've updated the title of this issue to make this a bit more clear.
